### PR TITLE
Dev 0.1.2

### PR DESCRIPTION
**Pull request by kamil-kaczmarek**
## API changes
**Step class**
* adapters.py removed
* postprocessing.py moved to the steppy-toolkit
* graph_info property renamed to the upstream_pipeline_structure
* cache_dirpath is now experiment_directory and is obligatory
* view_graph is now display_pipeline
* save_graph is now save_pipeline_as_png
* tmp is now cache
* adapter must be an instance of the class Adapter
* when you save something to disk use `persist` keyword, for example:
    * `save_as_png` is now `persist_to_disk`
* added assertions in the Step '__init__'
* changed docstrings
* added more logging info
* changed naming of multiple functions in Step

### CODE REVIEW

File: steppy/adapter.py [hunk 1]
```diff
@@ -14,10 +14,57 @@ class AdapterError(Exception):
...
+    """
     def __init__(self, adapting_recipes: Dict[str, AdaptingRecipe]):
+        """Adapter constructor
+
+        Note:
+            You have to import the extractor 'E' from this module to construct
+            adapters
+
+        Args:
+            adapting_recipes: Recipes used to control the input translation.
```
Comment (grzes314) [inline steppy/adapter.py]: I would add that `adapting_recipe` may be any literal and `E` is just a placeholder for values extracted from the outputs of previous steps.

File: steppy/adapter.py [hunk 2]
```diff
@@ -14,10 +14,57 @@ class AdapterError(Exception):
...
+                    extractors to the parent steps and combine the results
+                    into a dict with the same keys
+
+                5. Anything else: the value itself will be used as the argument
+                    to the transformer
+
+        """
         self.adapting_recipes = adapting_recipes
 
     def adapt(self, all_inputs: AllInputs) -> Dict[str, Any]:
```
Comment (grzes314) [inline steppy/adapter.py]: This one is on me, but I started to think that `all_inputs`, `AllInputs` aren't well chosen names -- they actually refer to the _outputs_ of previous steps.

File: steppy/base.py [hunk 3]
```diff
@@ -1,131 +1,132 @@
...
-    Step executes fit_transform on every step recursively starting from the very last step and making its way forward
-    through the input_steps. If step transformer was fitted already then said transformer is loaded in and the transform method
-    is executed.
-    One can easily debug the data flow by plotting the pipeline graph with either step.save_graph(filepath) method
-    or simply returning it in a jupyter notebook cell.
-    Every part of the pipeline can be easily accessed via step.get_step(name) method which makes it easy to reuse parts of the pipeline
-    across multiple solutions.
+    """Building block of steppy pipelines.
+
+    Step is an execution wrapper over the transformer (see BaseTransformer), which realizes single operation on data. With Step you can:
```
Comment (grzes314) [inline steppy/base.py]: Lines should be at most 100 characters long

File: steppy/base.py [hunk 4]
```diff
@@ -1,131 +1,132 @@
...
+        1. design multiple input/output data flows and connections between Steps.
+        2. handle persistence and caching of transformers and intermediate results.
+    Step executes 'fit_transform' method inspired by the sklearn on every step recursively starting from the very last Step and making its way forward through the input_steps.
+    One can easily debug the data flow by plotting the pipeline graph (see: step.persist_as_png(filepath) or return step in a jupyter notebook cell.
     """
     def __init__(self,
                  name,
                  transformer,
-                 input_steps=None,
+                 experiment_directory,
```
Comment (grzes314) [inline steppy/base.py]: This is why I like to do `def fun(self, *, arg1, arg2, arg3)`. Without it a small change like adding another argument can mess up a lot in code that uses this functions/class.

File: steppy/adapter.py (hunk 1 above)
Comment (kamil-kaczmarek) [reply steppy/adapter.py]: @grzes314 Ok, please add it.
@grzes314 can you also review all docstrings in `adapter.py` and add Example there? THX

File: steppy/adapter.py (hunk 2 above)
Comment (kamil-kaczmarek) [reply steppy/adapter.py]: @grzes314 agree, please refactor it. Also, make sure that examples and tests are fine.

File: steppy/base.py (hunk 4 above)
Comment (kamil-kaczmarek) [reply steppy/base.py]: So you recommend sth like:
`     def __init__(self, *,  name, transformer, ...):`

**Pull request merged**